userd/ui/ui.go

```diff
@@ -0,0 +1,53 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ui
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type UI interface {
+	// YesNo asks a yes/no question. The primary text
+	// will be printed in a larger font, the secondary text
+	// in the standard font and the (optional) footer will
+	// be printed in a small font.
+	//
+	// The value "true" is returned if the user clicks "yes",
+	// otherwise "false".
+	YesNo(primary, secondary string, options *Options) bool
+}
+
+type Options struct {
+	Footer  string
+	Timeout int
+}
+
+func New() (UI, error) {
+	switch {
+	case osutil.ExecutableExists("zenity"):
```

Comment:
Should we look also look at environment to determine which is preferred? I mean GNOME user would probably prefer zenity over kdialog. Opposite is probably true for a KDE user.

I see a couple of env variables that may be set: `XDG_CURRENT_DESKTOP`, `XDG_SESSION_DESKTOP`, `DESKTOP_SETTION` (this may be GNOME specific though).

Reply:
I second this opinion. The UI is hacky anyway but we can make it a little bit more pleasant.

Reply:
Thanks, I added this now.